Firestore/core/src/firebase/firestore/remote/serializer.cc

```diff
@@ -436,6 +446,19 @@ DocumentKey Serializer::DecodeKey(Reader* reader,
   return DocumentKey{std::move(local_path)};
 }
 
+DatabaseId Serializer::DecodeDatabaseId(
+    Reader* reader, const ResourcePath& resource_name) const {
+  if (resource_name.size() < 5 || resource_name[4] != "documents") {
```

Comment:
DatabaseIds can be parsed from database names which don't include the trailing "documents".

Database names are things like "/projects/foo/databases/bar"

Resource names are things like "/projects/foo/databases/bar/documents/collection/document"

Both constructs exist in the proto serialization so this shouldn't require "documents" exist here.